src/rgw/rgw_http_client.cc

```diff
@@ -832,16 +836,18 @@ int RGWHTTPManager::process_requests(bool wait_for_data, bool *done)
     CURLMsg *msg;
     while ((msg = curl_multi_info_read((CURLM *)multi_handle, &msgs_left))) {
       if (msg->msg == CURLMSG_DONE) {
-	CURL *e = msg->easy_handle;
-	rgw_http_req_data *req_data;
-	curl_easy_getinfo(e, CURLINFO_PRIVATE, (void **)&req_data);
-
-	long http_status;
-	curl_easy_getinfo(e, CURLINFO_RESPONSE_CODE, (void **)&http_status);
-
-	int status = rgw_http_error_to_errno(http_status);
-	int result = msg->data.result;
-	finish_request(req_data, status);
+        CURL *e = msg->easy_handle;
+        rgw_http_req_data *req_data;
+        curl_easy_getinfo(e, CURLINFO_PRIVATE, (void **)&req_data);
+        
+        curl_multi_remove_handle((CURLM *)multi_handle, e);
+
+        long http_status;
+        curl_easy_getinfo(e, CURLINFO_RESPONSE_CODE, (void **)&http_status);
+
+        int status = rgw_http_error_to_errno(http_status);
```

Comment:
what error code can we expect when curl chooses to abort a slow response?

Reply:
In the slow response, the http_status is still 200. Because the http_status is returned at the beginning of object downloading. When the response is too slow, the curl will return timeout error, and the msg->data.result is CURLE_OPERATION_TIMEDOUT(28).

Reply:
thanks. does that get mapped to EAGAIN so sync will retry?

Reply:
can you help me understand what a timeout here looks like to our sync coroutines? is there anything more that we need to do to recover/retry?